lib/internal/fs/glob.js

```diff
@@ -68,6 +87,22 @@ class Cache {
     this.#readdirCache.set(path, val);
     return val;
   }
+
+  async readdir(path) {
+    const cached = this.#readdirCache.get(path);
+    if (cached) {
+      return cached;
+    }
+    let val;
+    try {
+      val = await readdir(path, { __proto__: null, withFileTypes: true });
```

Comment:
use `opendir` instead of readdir

Reply:
and yield the results or just opendir? because yielding the result will cause incomplete cache

Reply:
then it probably makes sense to use `readdir` for the first iteration

Reply:
how will it solve the issue? on further iterations we should get the cache and not use `opendir` anyway...

Reply:
I meant first iteration of the feature